src/utils.hpp

```diff
@@ -91,6 +91,52 @@ std::string& to_cql_id(std::string& str);
 
 std::string& escape_id(std::string& str);
 
+inline size_t num_leading_zeros(cass_int64_t value) {
+  if (value == 0)
+    return 64;
+
+#if defined(_MSC_VER)
+  unsigned long index;
+#  if defined(_M_AMD64)
+  _BitScanReverse64(&index, value);
+#  else
+  // On 32-bit this needs to be split into two operations
+  char isNonzero = _BitScanReverse(&index, (unsigned long)(value >> 32));
+
+  if (isNonzero)
+    // The most significant 4 bytes has a bit set, and our index is relative to that.
+    // Add 32 to account for the lower 4 bytes that make up our 64-bit number.
+    index += 32;
+  else {
+    // Scan the last 32 bits by truncating the 64-bit value
+    _BitScanReverse(&index, (unsigned long) value);
+  }
+#  endif
+  // index is the (zero based) index, counting from lsb, of the most-significant 1 bit.
+  // For example, a value of 12 (b1100) would return 3. The 4th bit is set, so there are
+  // 60 leading zeros.
+  return 64 - index - 1;
+#else
+  return __builtin_clzll(value);
+#endif
+}
+
+inline cass_int64_t decode_zig_zag(cass_uint64_t n) {
+  // n is an unsigned long because we want a logical shift right
+  // (it should 0-fill high order bits), not arithmetic shift right.
+  return (n >> 1) ^ -(n & 1);
+}
+
+inline cass_uint64_t encode_zig_zag(cass_int64_t n) {
+  return (n << 1) ^ (n >> 63);
+}
+
+inline size_t varint_size(cass_int64_t value) {
+  // | with 1 to ensure magnitude <= 63, so (63 - 1) / 7 <= 8
+  size_t magnitude = num_leading_zeros(value | 1);
+  return magnitude ? (9 - ((magnitude - 1) / 7)) : 9;
```

Comment:
magnitude is an unsigned type, so subtracting 1 and then dividing by 7 causes it to be junk. We could cast magnitude to an int when doing the calculation instead of the branch...